File: 13. Rekursi Lanjut/D. Brankas Terkunci/d_brankas_terkunci.java
import java.util.Scanner;

public class Solution {
    public static int[] save = new int[17];
    public static boolean[] isVisited = new boolean[17];
    public static int N, K;

    public static void draw(int depth) {
        if (depth >= K - 1) {
            for (int i = 0; i < K; i++) {
                System.out.printf("%d ", save[i] + 1);
            }
            System.out.println();

        } else {
            for (int i = save[depth] + 1; i < N; i++) {
                if (!isVisited[i]) {
                    isVisited[i] = true;
                    save[depth + 1] = i;
                    draw(depth + 1);
                    isVisited[i] = false;
                }
            }
        }
    }

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        N = scan.nextInt();
        K = scan.nextInt();
        scan.close();

        for (int i = 0; i < N; i++) {
            save[0] = i;
            draw(0);
        }

    }
}

File: 13. Rekursi Lanjut/D. Brankas Terkunci/d_brankas_terkunci.go
package main

import (
	"fmt"
)

var save []int
var isVisited []bool
var N, K int

func draw(depth int) {
	if depth >= K-1 {
		for i := 0; i < K; i++ {
			fmt.Print(save[i]+1, " ")
		}
		fmt.Println()
	} else {
		for i := save[depth] + 1; i < N; i++ {
			if !isVisited[i] {
				isVisited[i] = true
				save[depth+1] = i
				draw(depth + 1)
				isVisited[i] = false

			}
		}
	}
}
func main() {
	save = make([]int, 17)
	isVisited = make([]bool, 17)

	fmt.Scan(&N, &K)
	for i := 0; i < N; i++ {
		save[0] = i
		draw(0)
	}
}
